tests/test_run_config_generator.py

```diff
@@ -0,0 +1,184 @@
+        expected_num_of_configs = expected_pa_configs * expected_model_configs
+
+        self._run_and_test_run_config_generator(
+            yaml_content, expected_config_count=expected_num_of_configs)
+
+    def test_default_config_two_models(self):
+        """
+        Test Default Two Models:
+            - No CLI options specified
+
+        log2(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY)+1 PA configs x
+        DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT+1 Model configs
+        will be generated by the auto-search
+        """
+
+        # yapf: disable
+        yaml_content = convert_to_bytes("""
+            profile_models:
+                - my-model0
+                - my-model1
+            """)
+        # yapf: enable
+
+        expected_pa_configs = len(
+            utils.generate_log2_list(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY))
+        expected_model_configs = DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT + 1
+        expected_num_of_configs = expected_pa_configs * expected_model_configs**2
+        self._run_and_test_run_config_generator(
+            yaml_content, expected_config_count=expected_num_of_configs)
+
+    def test_default_config_four_models(self):
+        """
+        Test Default Four Models:
+            - No CLI options specified
+
+        log2(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY)+1 PA configs x
+        DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT+1 Model configs
+        will be generated by the auto-search
+        """
+
+        # yapf: disable
+        yaml_content = convert_to_bytes("""
+            profile_models:
+                - my-model0
+                - my-model1
+                - my-model2
+                - my-model3
+            """)
+        # yapf: enable
+
+        expected_pa_configs = len(
+            utils.generate_log2_list(DEFAULT_RUN_CONFIG_MAX_CONCURRENCY))
+        expected_model_configs = DEFAULT_RUN_CONFIG_MAX_INSTANCE_COUNT + 1
+        expected_num_of_configs = expected_pa_configs * expected_model_configs**4
+
+        self._run_and_test_run_config_generator(
+            yaml_content, expected_config_count=expected_num_of_configs)
+
+    def _run_and_test_run_config_generator(self,
+                                           yaml_content,
+                                           expected_config_count,
+                                           pa_cli_args=None):
+        args = [
+            'model-analyzer', 'profile', '--model-repository', 'cli_repository',
+            '-f', 'path-to-config-file'
+        ]
+
+        if type(pa_cli_args) == list:
+            args = args + pa_cli_args
+        elif type(pa_cli_args) == str:
+            args.append(pa_cli_args)
+
+        protobuf = """
+            max_batch_size: 8
+            instance_group [
+            {
+                kind: KIND_CPU
+                count: 1
+            }
+            ]
+            """
+
+        self.mock_model_config = MockModelConfig(protobuf)
+        self.mock_model_config.start()
```

Comment:
what is this `MockModelConfig` for? and should there be a call to `.stop()` somewhere?

Reply:
Need to create a model config in the test. I've added the call to stop